game_modules/game_engine.py

```diff
@@ -0,0 +1,822 @@
+                    self.drawText(text, font, (255, 255, 255), (math.floor((self.windowWidth / 2)), math.floor((self.windowHeight / 2))))
+
+                    # Pelaaaja 2
+                    text = "Pelaaja 2 pisteet: " + str(self.score2)
+                    font = pygame.font.Font('fonts/OpenSans-Regular.ttf', 25)
+                    self.drawText(text, font, (255, 255, 255), (math.floor((self.windowWidth / 2)), math.floor((self.windowHeight / 1.7))))
+                    #Katsotaan kumpi pelaaja voitti
+                    if self.score1 > self.score2:
+                        text = "Pelaaja 1 voitti pelin!"
+                        font = pygame.font.Font('fonts/OpenSans-Regular.ttf', 40)
+                        self.drawText(text, font, (255, 255, 255), (math.floor((self.windowWidth / 2)), math.floor((self.windowHeight / 2.4))))
+                    elif self.score2 > self.score1:
+                        text = "Pelaaja 2 voitti pelin!"
+                        font = pygame.font.Font('fonts/OpenSans-Regular.ttf', 40)
+                        self.drawText(text, font, (255, 255, 255), (math.floor((self.windowWidth / 2)), math.floor((self.windowHeight / 2.4))))
+                #Nappien arvojen määrittäminen eri vaikeustasoille
+                if self.difficulty.name == "Easy" and self.gamemode.name == "Solo":
+                    fontsize = 30
+                    x = 60
+                    y = 220
+                    width = 280
+                    height = 75
+                    restart = "easy"
+                elif self.difficulty.name == "Easy" and self.gamemode.name == "Duel":
+                    fontsize = 25
+                    x = 60
+                    y = 250
+                    width = 280
+                    height = 50
+                    restart = "easy_duel"
+                elif self.difficulty.name == "Normal" and self.gamemode.name == "Solo":
+                    fontsize = 40
+                    x = 100
+                    y = 400
+                    width = 400
+                    height = 75
+                    restart = "normal"
+                elif self.difficulty.name == "Normal" and self.gamemode.name == "Duel":
+                    fontsize = 40
+                    x = 100
+                    y = 400
+                    width = 400
+                    height = 75
+                    restart = "normal_duel"
+                elif self.difficulty.name == "Hard" and self.gamemode.name == "Solo":
+                    fontsize = 40
+                    x = 200
+                    y = 400
+                    width = 400
+                    height = 75
+                    restart = "hard"
+                elif self.difficulty.name == "Hard" and self.gamemode.name == "Duel":
+                    fontsize = 40
+                    x = 200
+                    y = 400
+                    width = 400
+                    height = 75
+                    restart = "hard_duel"
```

Comment:
Gamemode and difficulty should be enums instead of comparing to string values as this makes refactoring harder -> Needless repetition